src/app/(privateProfile)/analytics/page.tsx

```diff
@@ -1,8 +1,349 @@
+	// 	},
+	// 	{
+	// 		id: 12,
+	// 		background: "--color-peach-pink",
+	// 		title: "Учебники",
+	// 		valuePercent: "2.00",
+	// 		valueRub: 2600.01,
+	// 	},
+	// 	{
+	// 		id: 13,
+	// 		background: "--color-forest-green",
+	// 		title: "Отпуск",
+	// 		valuePercent: "4.95",
+	// 		valueRub: 6437.57,
+	// 	},
+	// 	{
+	// 		id: 14,
+	// 		background: "--color-olive-green",
+	// 		title: "Театр",
+	// 		valuePercent: "1.30",
+	// 		valueRub: 1690.01,
+	// 	},
+	// 	{
+	// 		id: 15,
+	// 		background: "--color-pale-lemon",
+	// 		title: "Кредит",
+	// 		valuePercent: "20.00",
+	// 		valueRub: 26000.15,
+	// 	},
+	// 	{
+	// 		id: 16,
+	// 		background: "--color-dark-olive",
+	// 		title: "Страховка",
+	// 		valuePercent: "0.40",
+	// 		valueRub: 520.0,
+	// 	},
+	// 	{
+	// 		id: 17,
+	// 		background: "--color-golden-olive",
+	// 		title: "Киберспорт",
+	// 		valuePercent: "0.40",
+	// 		valueRub: 520.0,
+	// 	},
+	// 	{
+	// 		id: 18,
+	// 		background: "--color-light-blue-gray",
+	// 		title: "Путешествия",
+	// 		valuePercent: "0.40",
+	// 		valueRub: 520.0,
+	// 	},
+	// 	{
+	// 		id: 19,
+	// 		background: "--color-burnt-orange",
+	// 		title: "Кино",
+	// 		valuePercent: "0.40",
+	// 		valueRub: 520.0,
+	// 	},
+	// 	{
+	// 		id: 20,
+	// 		background: "--color-bright-lime",
+	// 		title: "Ипотека",
+	// 		valuePercent: "7.37",
+	// 		valueRub: 9586.33,
+	// 	},
+	// ];
+	const WINDOW_SIZE = 1440;
+	const MIN_ROW = 0;
+	const MAX_ROW = 8;
+
+	const [itemsToShow, setItemsToShow] = useState(MAX_ROW);
+	const [displayMode, setDisplayMode] = useState("rub");
+	const isEmptyPage = false;
+
+	useEffect(() => {
+		const handleResize = () => {
+			setItemsToShow(window.innerWidth <= WINDOW_SIZE ? MIN_ROW : MAX_ROW);
+		};
+
+		handleResize();
+
+		window.addEventListener("resize", handleResize);
+
+		return () => {
+			window.removeEventListener("resize", handleResize);
+		};
+	}, []);
+
+	const handleDisplayChange = (event: ChangeEvent<HTMLInputElement>) => {
+		setDisplayMode(event.target.value);
+	};
+
+	const data = {
+		labels: [
+			"Внезапная покупка", "Стрижка", "Бассейн", "Школа", "Еда", 
+			"Плата жилья", "Ногти", "Бензин", "Дорога работа", "Юрист", 
+			"Детский сад", "Учебники", "Отпуск", "Театр", "Кредит", 
+			"Страховка", "Киберспорт", "Путешествия", "Кино", "Ипотека"
+		],
+		datasets: [
+			{
+				label: "Расходы",
+				data: [
+					1300.01, 3900.02, 3250.02, 1638.83, 2652.06, 
+					15271.09, 390.0, 975.56, 1340.79, 9110.05, 
+					16192.09, 2600.01, 6437.57, 1690.01, 26000.15, 
+					520.0, 520.0, 520.0, 520.0, 9586.33
+				].map(value => displayMode === "rub" ? value : (value / 130000 * 100).toFixed(2)),
+				backgroundColor: [
+					"#21703c", "#b48dad", "#bfc3a8", 
+					"#eed6cd", "#b83400", "#c88b79", 
+					"#ff9d73", "#90b89e", "#c7b2af", 
+					"#8891a8", "#edc915", "#f8ab9e", 
+					"#489177", "#8f9e3e", "#eef6c6", 
+					"#8f670a", "#c0a30f", "#afb9cf", 
+					"#c66a04", "#dbed15"
+				],
```

Comment:
can you make util function which return necessary count of colors? 

For example:
backgroundColor: getRandomColors(20)